fuzzing/oss-fuzz-scripts/build.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
```

Comment:
> Assuming that, unlike the Apache 2.0 licensed files, these shell scripts are licensed exactly the same as the rest of the contents of the GitPython repository, it might be helpful to include a brief license header stating the license (BSD-3-Clause, if the same as the rest of GitPython), or even just an SPDX license header, any which would make this clear. Otherwise, especially if the Apache 2.0 license file is added in the `fuzzing` directory, readers might be misled.
> 
> I this might happen eventually even if not done now, so if there is a reason it should not be done, then that should be kept track of in some way.

Every file except the two we've discussed are my own creation and I defer to this project for choosing their license. If BSD-3-Clause is the preference just like the rest of GitPython, then BSD-3-Clause it is.

I have a few clarifying questions though:
1. Are the license comments applied in 5bef7ed7369e1d6f93161607e01dc02bcd1720ab good for me to copy verbatim, or did you have something else in mind?
1. Do we want the same header comments added to the dictionaries as well? We can certainly add them, but IMHO if they get to the point where anyone else may want to use them, they should probably be moved to https://github.com/gitpython-developers/qa-assets.
1. Should I add a call-out to the `fuzzing/README` or somewhere else asking any future contributors to add the headers as well?

Reply:
This is low priority because license sections of both the top-level and fuzzing readmes are clear that the project license applies to all but two particular files. The files are already BSD-3-Clause even without the headers. The BSD-3-Clause license also does not require headers and, for the kind of headers we're talking about, does not even require anyone to retain the headers once present.

When making a decision about this, I suggest also considering making a symlink in `fuzzing/` named `LICENSE-BSD` (or, if you prefer, `LICENSE-BSD-3-Clause`), pointing to `../LICENSE`.

> If BSD-3-Clause is the preference just like the rest of GitPython, then BSD-3-Clause it is.

Another possibility that comes to mind is that they could be dual-licensed as BSD-3-Clause and Apache-2.0, giving everyone a choice as to which to use, which would allow anyone who wants to derive their project's fuzz testing code from this one to refrain altogether from mentioning BSD-3-Clause it they're going to use Apache for the whole thing anyway.

To be clear, these two licenses are already compatible: material licensed under each can already be combined, even in the same file (the licenses and copyright notices for both, as well as additional information the Apache License requires, can be included). So there is no real need to dual-license. However, this would not be the only case where dual-licensing is used to give a choice between two permissive licenses. This is frequently done in Rust projects, which are often dual-licensed under the MIT License and Apache License 2.0. Then again, I wouldn't want to mislead anyone into thinking that even the two files that are Apache-2.0 only are dual-licensed.

I don't know what's best, but the stakes are low -- basically all the options are fine.

> Are the license comments applied in [5bef7ed](https://github.com/gitpython-developers/GitPython/commit/5bef7ed7369e1d6f93161607e01dc02bcd1720ab) good for me to copy verbatim, or did you have something else in mind?

The word "module" in the header makes sense for Python modules, but doesn't make sense for most other files such as shell scripts, so the wording should be adjusted to something like "This file" or "This script".

> Do we want the same header comments added to the dictionaries as well? We can certainly add them, but IMHO if they get to the point where anyone else may want to use them, they should probably be moved to https://github.com/gitpython-developers/qa-assets.

I think it is probably not needed there. GitPython tends to omit license headers for files where they are often omitted, such as Markdown files, `.gitignore` and `.gitattributes`, `Makefile`, requirements files, YAML files, and so forth. Perhaps most relevantly, none of the files in [`test/fixtures/`](https://github.com/gitpython-developers/GitPython/tree/main/test/fixtures) have them: some couldn't reasonably have them, while others could, and none do.

The top-level `.sh` scripts also don't have headers. In their case this seems like an oversight rather than intentional. But maybe that's a reason to not bother adding them to the new files--I hadn't been thinking about them when I recommended this. I still think they would be valuable, to reinforce that it is only two particular files in `fuzzing/` that are not under BSD-3-Clause. The other omissions I've just noticed are in any case a further reason that this definitely does not need to be addressed before the PR is merged. (Maybe they could be added both in `fuzzing/` and to those top-level scripts at the same time.)

Although there's a benefit to the headers in making it more obvious that, and how, people can incorporate the files (or parts of them) in their own work, many of the existing license headers were added just to address the inconsistency that some older files had headers while newer ones did not.

> Should I add a call-out to the `fuzzing/README` or somewhere else asking any future contributors to add the headers as well?

I think that should not be added.

- No such request is present in the top-level `README.md` or `CONTRIBUTING.md`.
- There isn't an established practice here, not even as much as I had thought (since the top-level shell scripts don't have them).
- If firm guidelines are ever decided on for this, then they can be checked on CI.

Reply:
Thanks as always for the super helpful reply! I'm going to mark this resolved since this was addressed in #1907 and I'm going over this PR's review comments to make sure I didn't miss anything.

Before I do I just wanted to leave my thoughts here regarding the dual-licensing portion of your comment, for posterity:

I think dual-licensing is an interesting idea, but probably more trouble than it's worth for the given state of things. Feel free to raise this again if you think its worth exploring more, but I like the less complex approach we've arrived at for now. I also think it's reasonable to consider replacing these Apache files completely if they end up being significantly refactored (for example, to address https://github.com/gitpython-developers/GitPython/pull/1901#discussion_r1565001609), which IMO would be a better investment.